pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -1655,6 +1704,64 @@ func updateBasedOnProfile(ctx *zedmanagerContext, oldProfile string) {
 	}
 }
 
+// countRunningAppsForUUIDs returns the number of app instances (from the provided uuidMap)
+// that are in a running state (BOOTING, RUNNING, HALTING, PURGING, or RESTARTING).
+func countRunningAppsForUUIDs(ctx *zedmanagerContext, uuidMap map[string]struct{}) (runningCount uint) {
+	sub := ctx.subAppInstanceStatus
+	items := sub.GetAll()
+	log.Noticef("countRunningAppsForUUIDs: %d items", len(items))
+	for _, s := range items {
+		status := s.(types.AppInstanceStatus)
+		log.Noticef("countRunningAppsForUUIDs: %s %s", status.UUIDandVersion, status.State)
+		// Check if this status belongs to one of the app UUIDs in our map.
+		if _, exists := uuidMap[status.UUIDandVersion.UUID.String()]; !exists {
+			continue
+		}
+		switch status.State {
+		case types.BOOTING, types.RUNNING, types.HALTING:
+			runningCount++
+		}
+	}
+	return runningCount
+}
+
+// Wait for all high-priority apps to start before launching this app.
+// Every 5 seconds, verify whether the priority apps are running.
+// If they have started, launch the app immediately; otherwise,
+// once the timeout expires, proceed to start the app regardless.
+func waitForActiveAppsThenStart(ctx *zedmanagerContext, uuids []string, config types.AppInstanceConfig) {
+	activeMap := make(map[string]struct{})
+
+	for _, uuid := range uuids {
+		activeMap[uuid] = struct{}{}
+	}
+	totalApps := len(activeMap)
+	log.Functionf("waitForAppsToStart: waiting for %d apps to start", totalApps)
+
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
+
+	timeoutCh := time.After(waitForAppsToStartTimeout)
+	startTime := time.Now()
+
+	for {
+		select {
+		case <-ticker.C:
+			runningCount := countRunningAppsForUUIDs(ctx, activeMap)
+			log.Functionf("waitForAppsToStart: %d/%d apps running, waited %v", runningCount, totalApps, time.Since(startTime))
+			if runningCount >= uint(totalApps) {
+				log.Functionf("waitForAppsToStart: all apps started after %v", time.Since(startTime))
+				handleCreateAppInstanceStatus(ctx, config)
+				return
+			}
+		case <-timeoutCh:
```

Comment:
I'm afraid this might be broken by applications with a delayed start set.
We have a feature that allows delaying the start of a given application. If it's enabled, the start can be delayed by an arbitrary amount of time, which may definitely exceed this timeout here.
To understand the logic, you can start looking here:
https://github.com/lf-edge/eve/blob/4d06d06b3be8bd428c9c0993cd8634e9c9607613/pkg/pillar/cmd/zedmanager/zedmanager.go#L604

Reply:
@kperakis-zededa I don't think we can enforce other apps to the delay of other Edge App just because this Edge App was active, so I would go with the option 2, just ignore apps with a startup delay....

Reply:
I would also check in which state we delay the start of the apps. If I'm not mistaken, applications are delayed *after* their resources have already been allocated (and that's the main point of this PR - to prioritize resources for apps that have already started).
So, I would consider treating apps in the `START_DELAYED` state as "running" in the new code.
But please double-check that I'm right and that this state indeed means all necessary resources are already reserved.

Reply:
And just a bit of background: the delayed start feature was introduced several years ago as a quick workaround for ordering application startup. If App A depends on App B, users typically set a delay for A equal to the expected start time of B plus some delta. This gives a sense of how the feature is actually used in practice.

Reply:
ok, then just add SATRT_DELAYED to the list of states you consider as "running" )